### PR TITLE
Fix PHP 8.1 deprecations

### PR DESCRIPTION
**Pull request by blankse**
Followup to https://github.com/pimcore/pimcore/pull/9862

Splitting from #10737 for branch 10.2 and extended it.

### CODE REVIEW

File: bundles/EcommerceFrameworkBundle/Model/AbstractPaymentInformation.php [hunk 1]
```diff
@@ -95,7 +95,14 @@ abstract public function getInternalPaymentId(): ?string;
...
+     * @return $this
      */
     abstract public function setInternalPaymentId(?string $internalPaymentId);
+
+    /**
+     * @param string|null $providerData
+     *
+     * @return $this
+     */
+    abstract public function setProviderData(?string $providerData);
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Model/AbstractPaymentInformation.php]: BC break?

Comment (blankse) [reply bundles/EcommerceFrameworkBundle/Model/AbstractPaymentInformation.php]: Can be, I changed it: https://github.com/pimcore/pimcore/pull/10964/commits/6cdefef040c0087a6665e6c663f4a62e5a172746

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@blankse thanks a lot!

**Comment by blankse** [comment]
@dvesh3 Nice :) Can you merge 10.2 to 10.x so I can merge it with https://github.com/pimcore/pimcore/pull/10737 ?

**Comment by dvesh3** [comment]
@blankse done 😉

**Pull request merged**